orchagent/txmonorch.cpp

```
@@ -0,0 +1,212 @@
+            } catch (...){
+                return;
+            }
+            if(periodToSet != m_pollPeriod){
+                setTimer(periodToSet);
+            }
+        } else {
+            SWSS_LOG_ERROR("Unexpected field %s", field.c_str());
+        }
+    }
+}
+
+void TxMonOrch::handleThresholdUpdate(const vector<FieldValueTuple>& data)
+{
+    for (auto element : data) {
+        const auto &field = fvField(element);
+        const auto &value = fvValue(element);
+
+        if (field == "value") {
+            try {
+                m_threshold = stoi(value);
+            } catch (...){
+                return;
+            }
+            SWSS_LOG_NOTICE("liora.debug m_threshold set [%u]", m_threshold);
+        } else {
+            SWSS_LOG_ERROR("Unexpected field %s", field.c_str());
+        }
+    }
+}
+
+bool TxMonOrch::createPortsMap()
+{
+    map<string, Port> &portsList =  gPortsOrch->getAllPorts();
+    for (auto &entry : portsList)
+    {
+        string name = entry.first;
+
+        Port p = entry.second;
+        if (p.m_type != Port::PHY)
+        {
+            continue;
+        }
+        string oidStr;
+        if (!m_countersPortNameTable.hget("", name, oidStr))
+        {
+            SWSS_LOG_ERROR("Failed to get port oid from counters DB");
+            return false;
+        }
+        m_portsMap.emplace(p.m_alias, oidStr);
+    }
+    return true;
+}
+
+void TxMonOrch::setPortsStateDb(string portAlias, PortState state)
+{
+    vector<FieldValueTuple> fieldValuesVector;
+    fieldValuesVector.emplace_back("port_state", stateNames[state]);
+    m_portsStateTable.set(portAlias, fieldValuesVector);
+}
+
+bool TxMonOrch::getTxErrCountersByAlias(string portAlias)
+{
+    string strValue;
+    string oidStr = m_portsMap.find(portAlias)->second;
+
+    if (!m_countersTable.hget(oidStr, "SAI_PORT_STAT_IF_OUT_ERRORS", strValue)) {
+        SWSS_LOG_ERROR("Failed to get port %s tx error counters from counters DB", oidStr.c_str());
```

it's better to throw an exception and catch it in the upper function